x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/aggs/InferencePipelineAggregator.java

```diff
@@ -0,0 +1,132 @@
+import org.elasticsearch.search.aggregations.pipeline.PipelineAggregator;
+import org.elasticsearch.search.aggregations.support.AggregationPath;
+import org.elasticsearch.xpack.core.ml.inference.results.InferenceResults;
+import org.elasticsearch.xpack.core.ml.inference.results.WarningInferenceResults;
+import org.elasticsearch.xpack.core.ml.inference.trainedmodel.InferenceConfigUpdate;
+import org.elasticsearch.xpack.ml.inference.loadingservice.Model;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+
+public class InferencePipelineAggregator extends PipelineAggregator {
+
+    private Map<String, String> bucketPathMap;
+    private InferenceConfigUpdate configUpdate;
+    private Model model;
+
+    public InferencePipelineAggregator(String name, Map<String,
+                                       String> bucketPathMap,
+                                       Map<String, Object> metaData,
+                                       InferenceConfigUpdate configUpdate,
+                                       Model model) {
+        super(name, bucketPathMap.values().toArray(new String[] {}), metaData);
+        this.bucketPathMap = bucketPathMap;
+        this.configUpdate = configUpdate;
+        this.model = model;
+    }
+
+    @SuppressWarnings({"rawtypes", "unchecked"})
+    @Override
+    public InternalAggregation reduce(InternalAggregation aggregation, InternalAggregation.ReduceContext reduceContext) {
+
+        InternalMultiBucketAggregation<InternalMultiBucketAggregation, InternalMultiBucketAggregation.InternalBucket> originalAgg =
+            (InternalMultiBucketAggregation<InternalMultiBucketAggregation, InternalMultiBucketAggregation.InternalBucket>) aggregation;
+        List<? extends InternalMultiBucketAggregation.InternalBucket> buckets = originalAgg.getBuckets();
+
+        List<InternalMultiBucketAggregation.InternalBucket> newBuckets = new ArrayList<>();
+        for (InternalMultiBucketAggregation.InternalBucket bucket : buckets) {
+            Map<String, Object> inputFields = new HashMap<>();
+
+            if (bucket.getDocCount() == 0) {
+                // ignore this empty bucket unless the doc count is used
+                if (bucketPathMap.containsKey("_count") == false) {
+                    newBuckets.add(bucket);
+                    continue;
+                }
+            }
+
+            for (Map.Entry<String, String> entry : bucketPathMap.entrySet()) {
+                String aggName = entry.getKey();
+                String bucketPath = entry.getValue();
+                Object propertyValue = resolveBucketValue(originalAgg, bucket, bucketPath);
+
+                if (propertyValue instanceof Number) {
+                    double doubleVal = ((Number) propertyValue).doubleValue();
+                    // NaN or infinite values indicate a missing value
```

Comment:
Another Note: NaN or infinite could also mean the agg actually generate a NaN/Infinite.  Unfortunately there's not a good way of knowing without inspecting the underlying agg itself and seeing if it's using its placeholder value.

Other pipelines don't particularly care, all NaN/Infinite are bad, so it's never been a problem.  But YMMV :)